src/components/button-toggle-group/button-toggle-group.js

```diff
@@ -0,0 +1,161 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import css from './../../utils/css';
+
+import InputLabel from './../../utils/decorators/input-label';
+import InputValidation from './../../utils/decorators/input-validation';
+
+import tagComponent from '../../utils/helpers/tags';
+
+/**
+ * An inoput group widget.
+ *
+ * == How to use an ButtonToggleGroup in a component:
+ *
+ * In your file
+ *
+ *   import ButtonToggleGroup from 'carbon/lib/components/button-toggle-group';
+ *
+ * To render an ButtonToggleGroup:
+ *
+ *   <ButtonToggleGroup validations=[]>
+ *     <ButtonToggle />
+ *   <ButtonToggleGroup />
+ *
+ * @class ButtonToggleGroup
+ * @constructor
+ * @decorators {InputLabel,InputValidation}
+ */
+const ButtonToggleGroup = InputLabel(InputValidation(
+class ButtonToggleGroup extends React.Component {
+  static propTypes = {
+    /**
+     * Children elements
+     *
+     * @property children
+     * @type {Node}
+     */
+    children: PropTypes.node.isRequired,
+
+    /**
+     * Custom className
+     *
+     * @property className
+     * @type {String}
+     */
+    className: PropTypes.string,
+
+    /**
+     * Selected value from children components
+     *
+     * @property value
+     * @type {String}
+     */
+    value: PropTypes.string
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (this.messageExists() && (nextProps.value !== this.props.value)) {
```

Comment:
is this not already handled by one of the decorators?
https://github.com/Sage/carbon/blob/master/src/utils/decorators/input-validation/input-validation.js#L170-L210

Reply:
Yes, but the one on the decorator was causing issues and not clearing the invalid state due to an issue with https://github.com/Sage/carbon/blob/master/src/utils/decorators/input-validation/input-validation.js#L186 always returning true.

Reply:
seems a bit dangerous to override a whole chunk of this method, could you override `_isCurrentlyActiveInput` instead or something?

Reply:
I already want to get rid of the duplicated logic. Definitely don't want to spread it through the code anymore